Describe internal conventions in CONTRIBUTING.md (#187)
myyoda/principles-paper: CONTRIBUTING.md

Where a convention serves a STAMPED principle, we say so, so that the repository practices what the paper preaches.

## Contents

- [Workflow and collaboration](#workflow-and-collaboration)
- [Prose and writing style](#prose-and-writing-style)
- [STAMPED terminology and principle references](#stamped-terminology-and-principle-references)
- [LaTeX and manuscript conventions](#latex-and-manuscript-conventions)
- [Tables and the checklist](#tables-and-the-checklist)
- [Principle section pattern](#principle-section-pattern)
- [Authorship and contributions](#authorship-and-contributions)
- [Bibliography](#bibliography)
- [Licensing (REUSE)](#licensing-reuse)
- [Build](#build)
- [Version control and data](#version-control-and-data)
- [Commit conventions](#commit-conventions)

## Workflow and collaboration

- All editing and discussion happens on GitHub: <https://github.com/stamped-principles/stamped-paper>.
- Changes land through pull requests.
  Do not edit `main` in place.
- Use GitHub issues for discussion.

>   Do not leave comments on Overleaf; an Overleaf mirror may exist for convenience, but GitHub is authoritative.

Reviewer: ```suggestion
  Do not leave comments, suggestions, or edits on Overleaf; an Overleaf mirror may exist for convenience, but GitHub is authoritative.
```

- A PR triggers a CI build of the manuscript PDF and posts a diff preview comment.
  Check it before requesting review.
- Keep posts on GitHub (issues, PR descriptions, comments) concise: lead with the point.
  Wrap long contextual material (logs, large snippets, more than ~10 lines) in a `<details>` block with a `<summary>` label so the prose stays readable.

> - In a PR description with a test plan, mark completed steps `- [x]` and pending steps `- [ ]`.

Reviewer: ```suggestion
- If the PR is not ready to merge, include a TODO section at the bottom of the PR description, mark completed steps `- [x]` and pending steps `- [ ]`.
```

  Refer to "STAMPED principles" rather than "STAMPED properties".
  Untangling the two senses in specific sentences is an ongoing refinement (see issue #178).
- **Term hierarchy: Research Object > Module > Component.**
  - A **research object** is a collection of data, code, and metadata that together represent the research as a complete unit.
  - A **module** is a separately Distributable unit of a research object (a subdataset, a collection of scripts, an environment definition).
  - A **component** is a trackable element that is part of a module (a file, a script, a config).
- **Workflow** is defined externally by WCI-FW and cited, not redefined by us.
- **Provenance** is the recorded history of how components were produced or modified.
- **Tool-agnostic framing.**
  STAMPED does not depend on any particular tool.
  Avoid phrases like "STAMPED-compliant tools"; prefer "tools complementary to STAMPED".
  In tool tables and examples, treat tools as illustrative, not prescriptive.
- Avoid the phrase "research object's results"; refer to the research object or its components directly.

### How to write a principle reference

> When referring to one of the seven principles in running prose, write the **capitalized name followed by its letter in parentheses**: "Tracking (T)", "Distributability (D)".

Reviewer: tildes for no line breaks there
```suggestion
When referring to one of the seven principles in running prose, write the **capitalized name followed by its letter in parentheses**: "Tracking~(T)", "Distributability~(D)".
```

- Manuscript text, bibliography, figures, and human-readable documentation are **CC-BY-4.0**.
- Code, tooling, and CI configuration are **Apache-2.0**.
- Vendored third-party helpers keep their own SPDX header (for example the MIT-licensed credit renderers).
- Every tracked file needs SPDX metadata.
  When you add a file, add an inline SPDX header where the format allows, or extend the matching `path` block in [`REUSE.toml`](REUSE.toml).
- Verify locally with `make reuse-lint` (or `uvx --from reuse reuse lint`).

## Build

- `make` (default) builds `main.pdf` and then fails if any citations or references are undefined.
- `make reuse-lint` checks licensing compliance.
- `make diagrams` renders Mermaid sources (`figures/*.mmd`) to SVG and PDF via `mermaid-cli`.
- `references.bib` is fetched from Zotero (see Bibliography).
- The PDF build, REUSE lint, and diff preview also run in CI on every PR.
  CI runs in disposable runners, exercising the repository's own Ephemerality and Portability.

## Version control and data

> - The repository is a DataLad dataset; large and binary files are managed with git-annex, while text files live in Git.

Reviewer: is this true? do we have large files in git annex somewhere?

Author: Good catch, the wording overstated it. Today only `arxiv.tar.gz` is annexed (per `git annex list`); the binary figure PDFs are in plain Git, since `.gitattributes`'s annex rule only applies to `datalad save` / `git annex add` and a plain `git add` bypasses it. Reworded to describe that mechanism rather than a current inventory (which would drift) in 0e8be38.
